src/java.base/share/classes/jdk/internal/constant/MethodTypeDescImpl.java

```diff
@@ -213,16 +213,25 @@ public String descriptorString() {
 
     @Override
     public MethodType resolveConstantDesc(MethodHandles.Lookup lookup) throws ReflectiveOperationException {
-        @SuppressWarnings("removal")
-        MethodType mtype = AccessController.doPrivileged(new PrivilegedAction<>() {
-            @Override
-            public MethodType run() {
-                return MethodType.fromMethodDescriptorString(descriptorString(),
-                                                             lookup.lookupClass().getClassLoader());
-            }
-        });
-
-        // let's check that the lookup has access to all the types in the method type
+        MethodType mtype;
+        try {
+            @SuppressWarnings("removal")
+            MethodType mt = AccessController.doPrivileged(new PrivilegedAction<>() {
+                @Override
+                public MethodType run() {
+                    return MethodType.fromMethodDescriptorString(descriptorString(),
+                        lookup.lookupClass().getClassLoader());
+                }
+            });
+            mtype = mt;
+        } catch (TypeNotPresentException ex) {
+            throw (ClassNotFoundException) ex.getCause();
```

Comment:
On a side note, I wonder if it's better to re-wrap the exception here as a `ReflectiveOperationException`, instead of just getting the cause. That will retain the entire stack trace.

Reply:
Fyi the exception was thrown at https://github.com/openjdk/jdk/blob/5a8af2b8b93672de9b3a3e73e6984506980da932/src/java.base/share/classes/sun/invoke/util/BytecodeDescriptor.java#L95.
I will make builds with both approaches, and see how their traces differ. I will stay on rethrow only if the original stacktrace is already informative enough.

Reply:
@JornVernee Here are a few traces for comparison: https://gist.github.com/5d441ab2159833e808303d1accb66ee8

In all cases, the entire stacktrace is retained; this ClassNotFoundException has the `MethodTypeDescImpl::resolveConstantDesc` in its trace already.

I believe directly unwrapping the `ClassNotFoundException` is the best:
1. In future optimization, we can parse the individual classes more directly (such as via `ClassDesc.resolveConstantDesc`) and the new code can just throw the CNFE directly without extra wrapping, as user don't anticipate wrapped causes.
2. `IllegalAccessException` throwing is done directly.

Reply:
Sorry, I've been out sick. Reviewed it now

Reply:
Hope you get better soon!